make: use code style CFLAGS from Makefile.cflags

gebart opened the pull request:
Since these flags are defined unconditionally in Makefile.cflags there's no reason to add them a second time in the board specific Makefiles.

kaspar030 commented on boards/arduino-mega2560/Makefile.include:
```diff
@@ -45,9 +45,9 @@ endif
 # define build specific options
 export CPU_USAGE = -mmcu=atmega2560
-export CFLAGS += -ggdb -g3 -std=gnu99 -Os -Wall -Wstrict-prototypes $(CPU_USAGE)
+export CFLAGS += -ggdb -g3 -Os $(CPU_USAGE)
```
Could you use CFLAGS_OPT for -Os?

kaspar030 replied:
(Sorry, might be too unrelated)

gebart replied:
I'll update avr in a separate commit, but this PR.

gebart commented:
@kaspar030 addressed comments and homogenized all CFLAGS settings for some platforms (cortexm_common, msp430_common, arduino-mega2560)

gebart commented:
(I don't really want to spend any time on the arm7 platforms because in my opinion they're old as crap and don't have much of a use case in today's IoT world)

OlegHahm commented:
Is there a reason to keep the variable at all?

gebart commented:
@OlegHahm I guess we can drop the CFLAGS_STYLE variable.

OlegHahm commented:
:+1: for dropping

gebart commented:
Dropped the CFLAGS_STYLE variable and updated msba2-common to follow other platforms more closely with regards to CFLAGS and the Makefile layout. (call it OCD or whatever, but it felt wrong leaving them out when updating the Makefiles..)

Added automatic selection of newlib-nano if available for msba2-common as well (all cortex platforms already use it if it is available)

gebart commented:
@OlegHahm was that an ACK?

OlegHahm commented:
Not yet.

OlegHahm commented:
Can someone test on the arduino-mega?

OlegHahm commented:
ACK for the rest.

OlegHahm commented:
@haukepetersen, don't you have one on your desk? Can you just try to flash default example and check if it still works? Then we can merge this.

kYc0o commented:
I just get one, I´ll test it asap

gebart commented:
rebased on top of latest master (after merging #4778)

OlegHahm commented:
@kYc0o, awesome, thanks!

kYc0o commented:
compiled and flashed default example correctly on arduino-mega2560

OlegHahm commented:
In this case, merge at will as soon as Travis agrees.

kYc0o commented:
Travis is not happy :/
```
periph_rtt:airfy-beacon:
Building application "periph_rtt" for "airfy-beacon" with MCU "nrf51".
/home/travis/build/RIOT-OS/RIOT/tests/periph_rtt/bin/airfy-beacon/periph_rtt.a(main.o): In function `cb':
/home/travis/build/RIOT-OS/RIOT/tests/periph_rtt/main.c:38: undefined reference to `rtt_get_counter'
/home/travis/build/RIOT-OS/RIOT/tests/periph_rtt/main.c:40: undefined reference to `rtt_set_alarm'
/home/travis/build/RIOT-OS/RIOT/tests/periph_rtt/bin/airfy-beacon/periph_rtt.a(main.o): In function `main':
/home/travis/build/RIOT-OS/RIOT/tests/periph_rtt/main.c:51: undefined reference to `rtt_init'
/home/travis/build/RIOT-OS/RIOT/tests/periph_rtt/main.c:54: undefined reference to `rtt_set_alarm'
collect2: error: ld returned 1 exit status
make: *** [all] Error 1
```

OlegHahm commented:
Let's see if we can make it happy by restarting.

gebart commented:
Travis is happy -> go
